Add a new kernel to model coupled annihilation of species: (#36)
idaholab/TMAP8: doc/content/source/kernels/ADMatCoupledDefectAnnihilation.md

# ADMatCoupledDefectAnnihilation

!syntax description /Kernels/ADMatCoupledDefectAnnihilation

## Overview

> This class adds to the residual a contribution from

Reviewer: ```suggestion
This kernel object adds to the residual a contribution from
```

Prefer using the MOOSE naming convention instead of the C++ one here.

\begin{equation}
\frac{dc}{dt} = \alpha K*(c_0-c)*v
\end{equation}

> where $c$ and $v$ are species concentrations provided as nonlinear or coupled variables, $\alpha$ is a constant coefficient, and $K$ is the annihilation rate defined as a material property, and $c_0$ is the equilibrium oncentration of the species described by $c$. $\alpha$ can be used to scale up the annihilation rate, for sensitivity analysis for example.

Reviewer: ```suggestion
where $c$ and $v$ are species concentrations provided as nonlinear or coupled variables, $\alpha$ is a constant coefficient, $K$ is the annihilation rate defined as a material property, and $c_0$ is the equilibrium concentration of the species described by $c$. $\alpha$ can be used to scale up the annihilation rate, which could be used during a sensitivity analysis.
```

Reviewer: Thanks for the extra context you added below this!
